client/src/components/FlashCards.jsx

```diff
@@ -54,15 +58,20 @@ const FlashCards = props => {
       }
       return { x, rot, scale, delay: undefined, config: { friction: 50, tension: down ? 800 : isGone ? 200 : 500 } };
     });
-    if (!down && gone.size === cardData.length)
+    if (!down && gone.size === cardData.length) {
+      if (mode === 'quiz') {
+        dispatch(actions.setQuizScore({ score, quizSize: cardData.length }));
+        setShowQuizScoreDialog(true);
+      }
       setTimeout(() => {
-        setScore(0);
         return gone.clear() || set(i => to(i));
       }, 600);
+    }
   });
   // Now we're just mapping the animated values to our view, that's it. Btw, this component only renders once. :-)
```

Comment:
I don't think you added this comment but even so when making changes could you just go through and remove it?

Reply:
done